packages/itwinui-react/src/core/utils/hooks/useControlledState.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+
+/**
+ * Wrapper over `useState` that always gives preference to the
+ * controlled state (which often comes from a prop).
+ *
+ * This is helpful when a component needs to support both uncontrolled
+ * and controlled states. If controlled value/setter is not passed,
+ * then it will work just like a regular `useState`.
+ *
+ * @example
+ * const [state, setState] = useControlledState(null, props.value, props.onChange);
+ */
+export const useControlledState = <T>(
+  initialValue: T,
+  controlledState: T,
+  setControlledState?: React.Dispatch<React.SetStateAction<T>>,
+) => {
+  const [uncontrolledState, setUncontrolledState] =
+    React.useState<T>(initialValue);
+
+  const state = controlledState ?? uncontrolledState;
```

Comment:
Just wondering, what happens if we do `useControlledState(10, null)`? In this case, we want to use the value `null` since `controlledState` takes preference over `uncontrolledState`/`initialValue`.

But `controlledState ?? uncontrolledState` = `null ?? 10` = `10` (and not `null`).

Is this expected behavior? Or should we use `uncontrolledState` instead of `controlledState` only when `controlledState` is `undefined` (not passed by the user, etc.)?

Reply:
sure, changed.

Reply:
Would be great is this can be clarified in the JSDocs too. Like what does the "preference" stand for ([here](https://github.com/iTwin/iTwinUI/pull/1537/files#diff-c9341964923594e472777ba3dbbaa4073dea52a6608a4456954eb1725f8d1c4bR8-R9)). Because preference could mean non-null, truthy, etc.